### ISSUE TITLE
[Bug] Font sizes on options page sometimes too small

### ISSUE THREAD

**Issue by irvinm** [piroor/treestyletab#3210]
## Abstract
It appears that sometimes options.html ends up rendering the fonts much smaller than intended.  I'm sure this is related to DPI scaling (4k monitor with OS @ 125% scaling) and\or a Firefox issue, but sometimes it works properly and other times is does not.  When it gets into this state, even restarting Firefox does not help and requires the OS to be rebooted.  [Notice that graphical items such as check-boxes radio buttons and such are not affected.]

I also wonder if this is related to some degree to the changes they had to make to get the scaling to work on the options page in this issue:  https://bugzilla.mozilla.org/show_bug.cgi?id=1398481 (Change was rolled out in FF86)

I have worked around this issue by forcefully using the following CSS in my userContent.css: (Medium I think is the intended size to be used, but I just tweaked to my liking and settled on 14px)

```
/* Force TST to use font-size "medium" ... sometimes reverted to small or x-small */
/* 14px works well on options page @ 110% in browser ... want group tabs at 110% */
@-moz-document url-prefix("moz-extension://fc4f9767-7140-437e-821d-392caf47413e/options/options.html") {
	body.independent {
		/* font-size: medium !important; */
		font-size: 14px !important;
	}
}
```

How it looks with fonts rendered too small:
![image](https://user-images.githubusercontent.com/979729/191998131-732f617d-7385-46c6-9977-2e577cc143b4.png)

With CSS workaround in place:
![image](https://user-images.githubusercontent.com/979729/191998308-357ab2a1-c6a5-4f8b-b25e-35e0eca8bbbb.png)

## Steps to reproduce

I don't think you will likely be able to reproduce this, but thought if you had any ideas on how to debug or information I can get from inspection when this happens I should be able to do that.

If not, I am ok with the workaround and we can just close this item.  It is obviously not a common situation in the community.

## Environment

 * Platform (OS): Windows 10 x64 (21H2)
 * Version of Firefox:  105.0 (although has been going on for a long time [post FF86] ... I always figured it was just a Firefox issue)
 * Version (or revision) of Tree Style Tab:  3.9.5

**Comment by piroor** [comment]
I've tested TST on Nightly 107.0a1 Windows 10 21H2 with 150% scaled display, but I couldn't see wrong text size...

It looks impossible to debug with addon debugger. The problem looks to happen inside C++ modules. Firefox looks to use [`GetDeviceCaps()`](https://learn.microsoft.com/en-us/windows/win32/api/wingdi/nf-wingdi-getdevicecaps) with `LOGPIXELSX`/`LOGPIXELSY` to calculate device pixel ratio, e.g. https://searchfox.org/mozilla-central/rev/ffa4d00965c5281def6d3ddcbcdf6259d38c9b9a/widget/windows/WinUtils.cpp#565 Logs with `MOZ_LOG=Widget:5` may give us something more helpful information.

**Comment by piroor** [comment]
If the information respond by `GetDeviceCaps()` is correct, we need to inspect more deeply through the code path around calculation of the font size.

**Comment by irvinm** [comment]
@piroor I tried to get the logging for widget, but was unsuccessful ... and I tried in a clean profile and on several machines, so I am guessing I am doing something wrong.

- On my Windows 10 box, I started Windows Sandbox and installed Firefox and TST
- I created a folder and opened a command prompt in that folder
- I launched Firefox with the following command:

"C:\Program Files\Mozilla Firefox\firefox.exe" -MOZ_LOG=Widget:5 -MOZ_LOG_FILE=Widget.log

- I navigated to the different TST options pages (built in and "more wide space" page)
- I closed Firefox
- I see that a set of files are created, but they are always 0 bytes.

I found another network logging example and repeated the above process with the following:

"c:\Program Files\Mozilla Firefox\firefox.exe" -MOZ_LOG=timestamp,rotate:200,nsHttp:5,cache2:5,nsSocketTransport:5,nsHostResolver:5,cookie:5 -MOZ_LOG_FILE=log.txt

In this case, I see that files are created and properly being logged.

Any ideas what I could try differently or what I am doing wrong?

![image](https://user-images.githubusercontent.com/979729/194919619-60160c7a-a3b2-422f-8917-31be75bb961b.png)

**Comment by piroor** [comment]
There are two logging functions https://searchfox.org/mozilla-central/rev/ffa4d00965c5281def6d3ddcbcdf6259d38c9b9a/widget/windows/WinUtils.cpp#67-68 and I've find them in the file but sadly codes for device pixel ratio don't call them... Thus we need to modify the C++ code locally to embed `LOG_D()` function calls and build Firefox. Here is the official document for Firefox developers:
https://firefox-source-docs.mozilla.org/setup/windows_build.html

**Comment by irvinm** [comment]
@piroor I am going to close this item.  Since I reported this item, I have been investigating the root cause of this issue and believe I have found it to be related to the paid utility "Groupy" from the company "Stardock".  I have found that I was seeing the same behavior that was being reported in this support thread:  [https://forums.stardock.com/506715/page/1/#3830109](https://forums.stardock.com/506715/page/1/#3830109 )

I figured that this behavior which happened most days had to be related to this issue with Firefox.  To test this theory, I disabled this tool and have been running over a week without any issues with the window titlebar text and\or Firefox addon text sizes.  (Although not using Groupy hasn't been fun)  

I will pursue a support avenue with them and worst case use the CSS workaround mentioned in the beginning.

I did also check into building Firefox and was able to do that successfully in a VM.  It wasn't as hard as I thought it might be.
